api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/MetricsServiceLifecycle.java

```diff
@@ -379,10 +384,16 @@ private int getDefaultTTL() {
     }
 
     private void initJobsService() {
+        RxSession rxSession = new RxSessionImpl(session);
         jobsService = new JobsServiceImpl();
         jobsService.setMetricsService(metricsService);
-        jobsService.setSession(new RxSessionImpl(session));
-        jobsService.setScheduler(new SchedulerImpl(new RxSessionImpl(session)));
+        jobsService.setSession(rxSession);
+        if (schedulerMode.equals("test")) {
```

Comment:
Would it be possible to overlay this class in the context of tests only? And not have it in the main jaxrs code base?